### ISSUE TITLE
deprecation: Is `frequency` key necessary in `lr_scheduler_config`?

### ISSUE THREAD

**Issue by adosar** [Lightning-AI/pytorch-lightning#20714]
### Outline & Motivation

The `lr_scheduler_config`  contains a `frequency` key as shown below:

```python
lr_scheduler_config = {
    # REQUIRED: The scheduler instance
    "scheduler": lr_scheduler,
    # The unit of the scheduler's step size, could also be 'step'.
    # 'epoch' updates the scheduler on epoch end whereas 'step'
    # updates it after a optimizer update.
    "interval": "epoch",
    # How many epochs/steps should pass between calls to
    # `scheduler.step()`. 1 corresponds to updating the learning
    # rate after every epoch/step.
    "frequency": 1,
    # Metric to monitor for schedulers like `ReduceLROnPlateau`
    "monitor": "val_loss",
    # If set to `True`, will enforce that the value specified 'monitor'
    # is available when the scheduler is updated, thus stopping
    # training if not found. If set to `False`, it will only produce a warning
    "strict": True,
    # If using the `LearningRateMonitor` callback to monitor the
    # learning rate progress, this keyword can be used to specify
    # a custom logged name
    "name": None,
}
```

The frequency of the learning rate decay is already handled by the `torch.optim.lr_scheduler`'s. For example, in `StepLR` this is handled by the `step_size` parameter as following:
```python
# Assuming optimizer uses lr = 0.05 for all groups
# lr = 0.05     if epoch < 30
# lr = 0.005    if 30 <= epoch < 60
# lr = 0.0005   if 60 <= epoch < 90
# ...
scheduler = StepLR(optimizer, step_size=30, gamma=0.1)
for epoch in range(100):
    train(...)
    validate(...)
    scheduler.step()
```
meaning that  ``{..., 'frequency': 2, ...}`` and ``step_size=2`` is equivalent to  ``{..., 'frequency': 1, ...}`` and ``step_size=4``.

Is there any case where the `frequency` key would be beneficial?

**Comment by davidgill97** [comment]
Yes it is useful for step-based training, where you have to specify validation frequency for scheduler like ReduceLROnPlateau.

**Comment by adosar** [comment]
Hi @davidgill97,

In the case of `ReduceLRonPlateau`, isn't `{..., 'frequency': 2, ...}` and `patience=2` equivalent to `{..., 'frequency': 1, ...}` and `patience=4`?

**Comment by davidgill97** [comment]
> Hi @davidgill97,
> 
> In the case of `ReduceLRonPlateau`, isn't `{..., 'frequency': 2, ...}` and `patience=2` equivalent to `{..., 'frequency': 1, ...}` and `patience=4`?

Not really. Scheduler is called every frequency steps, so if frequency is smaller than validation frequency, it will cause an error as it cant find the metric.

**Comment by SkafteNicki** [comment]
@adosar you are not wrong for saying that `frequency` is essentially serves the same purpose as setting `patience` or `step_size` or some other argument that controls how often to step.
A couple of arguments for keeping `frequency`:

* Backwards compatibility: it is not a required argument (default=1) so it is something users can just ignore if they decide to do so. Removing would mean that users would have to refactor their code.
* Name standardization: I understand the reason for calling it `patience` and `step_size` etc. in the different schedulers but this also means that if I want to change scheduler, then I need to both change the scheduler class and the arguments to the class.
* Reducing boilerplate: If you have written a custom scheduler and do not want to deal with this kind of logic of step frequency it can be offloaded to lightning.

**Comment by SkafteNicki** [comment]
Just for the sake of backwards compatibility, I am going to conclude that it is worth keeping the `frequency` key and therefore I will be closing the issue.
